### PR TITLE
Implement pipeline sync (update & disable jobs)

### PR DESCRIPTION
**Pull request by d2lam**
Update and disable jobs in sync when necessary

### CODE REVIEW

File: lib/pipeline.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint no-param-reassign: ["error", { "props": false }] */
```
Comment (stjohnjohnson) [inline lib/pipeline.js:1]: Why are we disabling this just in this file?

File: lib/pipeline.js [hunk 2]
```diff
@@ -47,24 +47,44 @@ class PipelineModel extends BaseModel {
...
+                    const parsedConfigJobNames = Object.keys(parsedConfig.jobs);
+
+                    // Loop through all existing jobs
+                    existingJobs.forEach(job => {
+                        // if it's in the yaml, update it
+                        if (parsedConfigJobNames.includes(job.name)) {
+                            job.containers = parsedConfig.jobs[job.name].map(j => j.image);
+                            jobsToSync.push(job.update());
+                        // if it's not in the yaml, disable it
+                        } else if (!/^PR\-/.test(job.name)) {
```
Comment (stjohnjohnson) [inline lib/pipeline.js]: We should just expose a method in `Job` called `.isPR()`

Comment (d2lam) [inline lib/pipeline.js]: We could

File: lib/pipeline.js (hunk 1 above)
Comment (d2lam) [inline lib/pipeline.js:1]: This is because of line 60 and 64, to update the job. We don't do this for other files.

**Comment by d2lam** [comment]
@stjohnjohnson  Added .isPR() function to Job model

File: lib/pipeline.js (hunk 1 above)
Comment (stjohnjohnson) [inline lib/pipeline.js:1]: I'd LOVE to be able to do this, but it's in our eslint rules.  If we want to do this, let's remove that rule globally, not per file.

**Comment by stjohnjohnson** [comment]
👍 
mrw_i_dodged_yet_another_pre-ordered_video_game_disaster

**Pull request merged**